treeherder/etl/perf_data_adapters.py

```diff
@@ -71,61 +71,79 @@ def __init__(self):
                         "metadata": {"type": "object"}  # (holds 'options' from talos data & various auxiliary data including 'test_aux', 'talox_aux', 'results_aux', and 'results_xperf')
                     },
                     "required": [
-                        "date", "series_signature", "replicates", "testsuite",
-                        "test"
+                        "date", "series_signature", "testsuite",
                     ]
                 }
             },
             "required": ["blob", "job_guid", "name", "type"]
         }
 
-    def calculate_series_data(
-            self, job_id, result_set_id, push_timestamp, replicates):
+
+    @staticmethod
+    def _round(num):
+        # Use a precision of .2f for all numbers stored
+        # to insure we don't have floats with giant mantissa's
+        # that inflate the size of the stored data structure
+        return round(num, 2)
+
+
+    @staticmethod
+    def _calculate_summary_data(job_id, result_set_id, push_timestamp,
+                               results):
+        values = []
+        for test in results:
+            values += results[test]
+
+        if values:
+            geomean = math.exp(sum(map(math.log, values))/len(values))
```

Comment:
Ok, didn't understand the explanation completely 100%, but it sounds like the graphserver formula is what we want. Will change.